### PR TITLE
feat(jupyter): add fonts as assets to service

### PR DESCRIPTION
**Pull request by saffaalvi**
Closes https://github.com/StatCan/kubeflow/issues/50

Fonts for the `crud-web-app/jupyter-web-app` component are loaded from `https://fonts.googleapis.com/`. We have experienced some issues with page load being blocked, waiting on `fonts.gstatic.com`.

We decided to move all the fonts into the service so issues with external services don't affect ours.

This PR creates a new `fonts` folder in `components/crud-web-apps/jupyter/frontend/src/assets/` containing all the `.woff2` files for the fonts used. Those files are then imported in `components/crud-web-apps/jupyter/frontend/src/styles.scss` using `fonts.css`.

### CODE REVIEW

**Comment by DavidSpek** [comment]
@saffaalvi Can you sign the Google CLA so your PRs won't get hung up on that?

**Comment by saffaalvi** [comment]
@DavidSpek I am currently going through StatCan's internal process for signing the CLA so hopefully that will be done soon. I apologize for the delay.

**Comment by saffaalvi** [comment]
@googlebot I signed it

**Comment by saffaalvi** [comment]
@DavidSpek I signed the CLA, is there anything else needed from my end to move along with testing this PR?

**Comment by DavidSpek** [comment]
@wg102 Could you maybe rebase and rework this PR as the files have changed slightly since this PR was created.

**Comment by wg102** [comment]
@DavidSpek rebased and squashed

**Comment by kimwnasptd** [comment]
@saffaalvi apologies for the late review. I don't have an issue with including the fonts in the compilation process with sass, since it will also ensure the apps work as expected in air-gapped environments. My main review comment is that I'd like us to have the fonts only in one place for all the web apps.

With that in mind I propose the following changes in this PR:
1. Move the fonts files and font-face rules to the common library
    1. place the `fonts.css` file to `kubeflow-common-lib/projects/kubeflow/src/lib/fonts.scss`
    2. extend the `copyCSS` rule in `kubeflow-common-lib/package.json` to also copy the new `fonts.scss` file. You can extend it with this command `&& cp ./projects/kubeflow/src/lib/fonts.scss ./dist/kubeflow/lib`
    3. move the `fonts` directory to `kubeflow-common-lib/projects/kubeflow/src/assets/fonts`
2. Update the three web apps we have to `@import` the `fonts.scss` file from the created module
    1. modify the `{jupyter/volumes/tensorboards}/frontend/package.json` file and add the `-r` arg in the `copyLibAssets` npm rule of each app
    2. update the `styles.scss` file of each app to `@import '~kubeflow/lib/fonts.scss';`

**Comment by kimwnasptd** [comment]
We've also updated our CI to run format checks before each PR. Could you do one last rebase on top of the latest master to ensure that this PR will trigger the correct tests?

**Comment by wg102** [comment]
Hi @kimwnasptd I am taking over for @saffaalvi. Thanks for the review, will do, having it in common would be better indeed.
I was wondering is there a reason the fonts should be in `kubeflow-common-lib/projects/kubeflow/src/lib/` instead of `kubeflow-common-lib/projects/kubeflow/src/assets/`? 
Similarly why put the `font.scss` file  in `/src/lib `and not directly `src`?

**Comment by saffaalvi** [comment]
@googlebot I consent.

**Comment by wg102** [comment]
@googlebot I consent

**Comment by wg102** [comment]
I am trying to get the https://github.com/kubeflow/kubeflow/issues/5869 fixed here as well, since we lose the logos with the fonts in assets. This is not yet in the PR.

**Comment by kimwnasptd** [comment]
> I am trying to get the #5869 fixed here as well, since we lose the logos with the fonts in assets. This is not yet in the PR.

Don't worry about this one @wg102, I've pushed a PR and will help with it in parallel https://github.com/kubeflow/kubeflow/pull/5942.

> I was wondering is there a reason the font.scss file should be in /src/lib and not src?

I'd like to keep only top level files in the `src` directory, either in the apps or in the library. Files like the `index.html` or the global `styles.scss` and from them we will include all other sub-files that will be living alongside the rest of the source code.

One future improvement we could do would be to create a distinct folder for our SCSS files under the `src/lib dir, so that they are more organized. But this can happen in a distinct future PR.

File: components/crud-web-apps/jupyter/frontend/src/index.html [hunk 1]
```diff
@@ -6,10 +6,6 @@
     <base href="/" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <link rel="icon" type="image/x-icon" href="favicon.ico" />
-    <link
```
Comment (kimwnasptd) [inline components/crud-web-apps/jupyter/frontend/src/index.html:9]: Could you also remove this import from the volumes and tensorboards web apps as well?

**Comment by kimwnasptd** [comment]
@wg102 I noticed that the last commit introduced the fonts files in each web app, which is expected since the `npm run build:watch` command copies them there. But we should not track these files with git and only have them live in the common code.

Could you also add a commit that updates the `frontend/.gitignore` files of the three web apps to not include the `/src/assets/fonts` directories respectively? If you then rebase and put this as the first commit of your branch then I believe it should automatically not include the fonts in the app's code, in the other commits.

**Comment by kimwnasptd** [comment]
@wg102 I see you've implemented all of my review comments, thanks! You can also give a small heads up once you've implemented a review change so that a reviewer can, timely, re-evaluate the updated code.

We will need to update the [Katib](https://github.com/kubeflow/katib/tree/master/pkg/new-ui/v1beta1) and [Models](https://github.com/kubeflow/kfserving/tree/master/web-app) web apps as well to do a `cp -r` when copying the assets, in the `package.json` files. I can take care of these.

Thank your for your PR and time!

/lgtm
/approve

**Comment by DavidSpek** [comment]
@kimwnasptd Now that this PR has been merged and I don't think there are any other big frontend ones open, should we now do the Angular upgrade? I'd like to time this fairly quickly to avoid problems with PRs people might open in the mean time.

**Comment by DavidSpek** [comment]
Forgot that https://github.com/kubeflow/kubeflow/pull/5880 should probably be merged before doing the upgrade.

**Pull request merged**